xilem/examples/calc.rs

```diff
@@ -0,0 +1,323 @@
+        // 7 8 9 X
+        flex_row((
+            digit_button("7").flex(1.),
+            digit_button("8").flex(1.),
+            digit_button("9").flex(1.),
+            operator_button(MathOperator::Multiply).flex(1.),
+        ))
+        .flex(1.0),
+        // 4 5 6 -
+        flex_row((
+            digit_button("4").flex(1.),
+            digit_button("5").flex(1.),
+            digit_button("6").flex(1.),
+            operator_button(MathOperator::Subtract).flex(1.),
+        ))
+        .flex(1.0),
+        // 1 2 3 +
+        flex_row((
+            digit_button("1").flex(1.),
+            digit_button("2").flex(1.),
+            digit_button("3").flex(1.),
+            operator_button(MathOperator::Add).flex(1.),
+        ))
+        .flex(1.0),
+        // bottom row
+        flex_row((
+            expanded_button("±", |data: &mut Calculator| data.negate()).flex(1.),
+            digit_button("0").flex(1.),
+            digit_button(".").flex(1.),
+            expanded_button("=", |data: &mut Calculator| data.on_equals()).flex(1.),
+        ))
+        .flex(1.0),
+    ))
+    .gap(GRID_GAP)
+    .cross_axis_alignment(CrossAxisAlignment::Fill)
+    .main_axis_alignment(MainAxisAlignment::End)
+    .must_fill_major_axis(true)
+}
+
+/// Creates a horizontal centered flex row designed for the display portion of the calculator.
+pub fn centered_flex_row<Seq, Marker>(sequence: Seq) -> Flex<Seq, Marker> {
+    flex(sequence)
+        .direction(Axis::Horizontal)
+        .cross_axis_alignment(CrossAxisAlignment::Center)
+        .main_axis_alignment(MainAxisAlignment::Start)
+        .gap(5.)
+}
+
+/// Creates a horizontal filled flex row designed to be used in a grid.
+pub fn flex_row<Seq, Marker>(sequence: Seq) -> Flex<Seq, Marker> {
+    flex(sequence)
+        .direction(Axis::Horizontal)
+        .cross_axis_alignment(CrossAxisAlignment::Fill)
+        .main_axis_alignment(MainAxisAlignment::SpaceEvenly)
+        .gap(GRID_GAP)
+}
+
+/// Returns a label intended to be used in the calculator's top display.
+/// It sets the text size to DISPLAY_FONT_SIZE.
+fn display_label(text: &str) -> impl WidgetView<Calculator> {
+    label(text).text_size(DISPLAY_FONT_SIZE)
+}
+
+/// Returns a button contained in an expanded box. Useful for the buttons so that
+/// they take up all available space in flex containers.
+fn expanded_button(
+    text: &str,
+    callback: impl Fn(&mut Calculator) + Send + Sync + 'static,
+) -> impl WidgetView<Calculator> + '_ {
+    sized_box(button(text, callback)).expand()
+}
+
+/// Returns an expanded button that triggers the calculator's operator handler,
+/// on_entered_operator().
+fn operator_button(math_operator: MathOperator) -> impl WidgetView<Calculator> {
+    expanded_button(math_operator.as_str(), move |data: &mut Calculator| {
+        data.on_entered_operator(math_operator);
+    })
+}
+
+/// Returns an expanded button that triggers the calculator's digit handler,
+/// on_entered_digit().
```

Comment:
This comment is just re-stating the code, which doesn't really add much
```suggestion
/// A button which adds `digit` to the current input when pressed
```